base/Base.jl

```diff
@@ -405,6 +408,7 @@ function __init__()
     init_depot_path()
     init_load_path()
     init_active_project()
+    append!(_sysimage_modules, keys(loaded_modules))
```

Comment:
I think we need `empty!` it for the incremental builds.

```suggestion
    append!(empty!(_sysimage_modules), keys(loaded_modules))
```